backend/src/test/java/com/github/redstylzz/backend/service/CategoryServiceTest.java

```diff
@@ -78,20 +81,44 @@ void shouldReturnListOnSuccessfulRename() {
         when(repository.findByUserIDAndCategoryID(anyString(), anyString())).thenReturn(TestDataProvider.testCategory());
         when(repository.existsByUserIDAndCategoryName(anyString(), anyString())).thenReturn(false);
 
-        underTest.renameCategory(user, "", "");
+        List<Category> wantedList = underTest.renameCategory(user, "", "");
 
         verify(repository).save(any(Category.class));
         verify(repository).findAllByUserID(anyString());
-
+        assertEquals(List.of(), wantedList);
     }
 
     @Test
     void shouldReturnListOnSuccessfulDelete() {
         MongoUser user = TestDataProvider.testUser();
 
-        underTest.deleteCategory(user, "");
+        List<Category> wantedList = underTest.deleteCategory(user, "");
 
         verify(repository).deleteByCategoryID(anyString());
+        verify(paymentRepo).deleteAllByUserIDAndCategoryID(anyString(), anyString());
         verify(repository).findAllByUserID(anyString());
+        assertEquals(List.of(), wantedList);
+    }
+
+    @Test
+    void shouldSaveCategory() {
+        String userID = testUser().getId();
+        String categoryID = testCategory().getCategoryID();
+        BigDecimal paymentSum = BigDecimal.ZERO;
+        when(repository.findByUserIDAndCategoryID(anyString(), anyString())).thenReturn(testCategory());
+
+        underTest.setCategorySum(userID, categoryID, paymentSum);
+
+        verify(repository).save(any(Category.class));
```

Comment:
a verify statement should ideally never include an `any()` method, because you want to verify that the exact method call you expected was carried out, i.e. including the correct category here